src/scheduler/limits.cpp

```diff
@@ -912,43 +898,34 @@ check_limits(server_info *si, queue_info *qi, resource_resv *rr, schd_error *err
 					if (te->event_type == TIMED_RUN_EVENT) {
 						if (svr_counts != NULL) {
 							cts = find_alloc_counts(svr_counts->user, te_rr->user);
-							if (svr_counts->user == NULL)
-								svr_counts->user = cts;
 							update_counts_on_run(cts, te_rr->resreq);
-							svr_counts_max->user =
-								counts_max(svr_counts_max->user, cts);
-							if (svr_counts_max->user == NULL) {
-								error = true;
+							counts_max(svr_counts_max->user, cts);
+							if (svr_counts_max->user.size() == 0) {
+								error = 1;
 								break;
 							}
 
 							cts = find_alloc_counts(svr_counts->group, te_rr->group);
-							if (svr_counts->group == NULL)
-								svr_counts->group = cts;
 							update_counts_on_run(cts, te_rr->resreq);
-							svr_counts_max->group =
-								counts_max(svr_counts_max->group, cts);
-							if (svr_counts_max->group == NULL) {
-								error = true;
+							counts_max(svr_counts_max->group, cts);
+							if (svr_counts_max->group.size() == 0) {
+								error = 1;
 								break;
 							}
 
 							cts = find_alloc_counts(svr_counts->project, te_rr->project);
-							if (svr_counts->project == NULL)
-								svr_counts->project = cts;
 							update_counts_on_run(cts, te_rr->resreq);
-							svr_counts_max->project =
-								counts_max(svr_counts_max->project, cts);
-							if (svr_counts_max->project == NULL) {
-								error = true;
+							counts_max(svr_counts_max->project, cts);
+							if (svr_counts_max->project.size() == 0) {
+								error = 1;
 								break;
 							}
 
-							update_counts_on_run(svr_counts->all, te_rr->resreq);
-							svr_counts_max->all =
-								counts_max(svr_counts_max->all, svr_counts->all);
-							if (svr_counts_max->all == NULL) {
-								error = true;
+							cts = find_alloc_counts(svr_counts->all, std::string(PBS_ALL_ENTITY));
+							update_counts_on_run(cts, te_rr->resreq);
```

Comment:
You can pass the char * PBS_ALL_ENTITY in.  C++ will do the conversion.  The same thing will happen though.  Here you are just being more explicit. (here and other places)